python/housinginsights/ingestion/LoadData.py

```diff
@@ -286,8 +289,267 @@ def rebuild(self):
+
+            api_results = self._items_divide(numerator, denominator)
+            api_results['data_id'] = data_id
+
+        # If data_id isn't one of our custom ones, assume it is a column name
+        else:
+            api_results = self._get_weighted_census_results(grouping, data_id)
+
+            # transform items to dict with group/value instead of list of dict
+            temp_items = {}
+            for item in api_results['items']:
+                temp_items[item['group']] = item['value']
+
+            api_results['items'] = temp_items
+
+        return api_results
+
+    def _get_weighted_census_results(self, grouping, field):
+        """
+        Queries the census table for the relevant field and returns the results
+        as a weighted count returns the standard 'items' format.
+
+        Currently only implemented for the 'counts' weighting factor not for
+        the proportion version.
+        """
+        with self.engine.connect() as conn:
+            q = "SELECT census_tract, {field} FROM census".format(
+                field=field)  # TODO need to add 'year' column for multiple census years when this is added to the data
+            proxy = conn.execute(q)
+            census_results = [dict(x) for x in proxy.fetchall()]
+
+            # Transform the results
+            items = []  # For storing results as we go
+
+            if grouping == 'census_tract':
+                # No weighting required, data already in proper format
+                for r in census_results:
+                    output = dict({'group': r['census_tract'],
+                                   'value': r[field]})
+                    items.append(output)
+
+            elif grouping in ['ward', 'neighborhood_cluster']:
+                proxy = conn.execute(
+                    "SELECT DISTINCT {grouping} FROM census_tract_to_{grouping}".format(
+                        grouping=grouping))
+                groups = [x[0] for x in proxy.fetchall()]
+
+                for group in groups:
+                    proxy = conn.execute(
+                        "SELECT * FROM census_tract_to_{grouping} WHERE {grouping} = '{group}'".format(
+                            grouping=grouping, group=group))
+                    results = [dict(x) for x in proxy.fetchall()]
+
+                    count = 0
+                    for result in results:
+                        tract = result['census_tract']
+                        factor = result['population_weight_counts']
+                        matching_data = next((item for item in census_results if
+                                              item["census_tract"] == tract),
+                                             {field: None})
+                        if matching_data[field] is None:
+                            logging.warning(
+                                "Missing data for census tract when calculating weightings: {}".format(
```

Comment:
Would be helpful to add which data field was missing in addition to which tract. When I ran this locally, I got 34 of these warnings but don't know whether to be worried about the mssing data or not.